engine/src/main/scala/cromwell/webservice/WorkflowQueryPagination.scala

```diff
@@ -0,0 +1,41 @@
+package cromwell.webservice
+
+import cromwell.services.MetadataServiceActor.QueryMetadata
+import spray.http.HttpHeaders.Link
+import spray.http.{HttpHeader, Uri}
+
+
+trait WorkflowQueryPagination {
+
+  protected def generatePaginationParams(page: Int, pageSize: Int): String = {
+    s"page=$page&pagesize=$pageSize"
+  }
+
+  //Generates link headers for pagination navigation https://tools.ietf.org/html/rfc5988#page-6
+  protected def generateLinkHeaders(uri: Uri, metadata: Option[QueryMetadata]): Seq[HttpHeader] = {
+    //strip off the query params
+    val baseUrl = uri.scheme + ":" + uri.authority + uri.path
+    metadata match {
+      case Some(meta) =>
+        (meta.page, meta.pageSize) match {
+          case (Some(p), Some(ps)) =>
+
+            val firstLink = Link(Uri(baseUrl).withQuery(generatePaginationParams(1, ps)), Link.first)
```

Comment:
Also, what headers does it return for the POST version of the same endpoint, used by Firecloud (I believe)?

The POST endpoint reads parameters in the body, not the query parameters, so I'm not exactly sure what the proper link headers are in that case, as the expected header responses for GET nor POST aren't listed in the `CromwellApiServiceSpec` nor the README.

If the headers are working for users in hotfix, I don't think we need to change anything there. But the generated headers here in develop appear to need some help.